Extensions/PlatformBehavior/platformruntimebehavior.ts

```diff
@@ -42,11 +46,32 @@ namespace gdjs {
       return runtimeScene.platformsObjectsManager;
     }
 
+    /**
+     * Update the AABB given to the spacial search of every platforms.
+     * It's only done once per frame.
+     * @param runtimeScene
+     */
+    updateAABBIfNeeded(runtimeScene: gdjs.RuntimeScene) {
+      const time = runtimeScene.getTimeManager().getTimeFromStart();
```

Comment:
I guess you're using the time as a way to check if it's worth updating the AABB... but the time will always be the same during a frame, so how is this useful exactly?

Reply:
Ok I actually understand that this is useful because as you said "The doStepPreEvents call order does not impose Platforms AABB to be updated before that platformer objects doStepPreEvents are called.".
I wonder though if there is another way to do it.

Reply:
I took a look at the call order in renderAndStep
* _updateObjectsPreEvents
* _eventsFunction
* _updateObjectsPostEvents
* render

Doing the platform AABB update in doStepPostEvents should work because it happens between event and doStepPreEvents. There is still the case when the platformer object is a platform but it can be updated without using a Set.

In case of unit tests, the tests modify the objects before _updateObjectsPreEvents, doesn't it? If so, I think that tests should modify the objects at the time events would.
For assertions, I find it important to be able to check what the events get in addition to what is displayed on the screen. What do you think about a TestRuntimeScene that split renderAndStep in these 2 methods:
* updateObjectsPreEvents(elapsedTime: float)
* updateObjectsPostEventsAndRender(elapsedTime: float)

Reply:
I updated my previous message because it was not very understandable.
Do you think this solution would be better than what is implemented in this PR?
Did you had another solution in mind?